src/abstractdataframe/io.jl

```diff
@@ -164,128 +165,150 @@ function html_escape(cell::AbstractString)
-        for j in 1:mxcol
-            s = html_escape(ct[j])
-            title = html_escape(ct_title[j])
-            write(io, "<th title=\"$title\">$s</th>")
+
+        mxcol = tryparse(Int, get(ENV, "DATAFRAMES_COLUMNS", "100"))
+
+        if isnothing(mxcol)
+            mxcol = 100
         end
-        write(io, "</tr>")
+    else
+        mxrow = -1
+        mxcol = -1
     end
-    write(io, "</thead>")
-    write(io, "<tbody>")
-    for row in 1:mxrow
-        write(io, "<tr>")
-        if rowid === nothing
-            write(io, "<th>$row</th>")
-        else
-            write(io, "<th>$rowid</th>")
-        end
-        for column_name in cnames
-            if isassigned(df[!, column_name], row)
-                cell_val = df[row, column_name]
-                if ismissing(cell_val)
-                    write(io, "<td><em>missing</em></td>")
-                elseif cell_val isa Markdown.MD
-                    write(io, "<td>")
-                    show(io, "text/html", cell_val)
-                    write(io, "</td>")
-                elseif cell_val isa SHOW_TABULAR_TYPES
-                    write(io, "<td><em>")
-                    cell = sprint(ourshow, cell_val, 0)
-                    write(io, html_escape(cell))
-                    write(io, "</em></td>")
-                else
-                    cell = sprint(ourshow, cell_val, 0)
-                    write(io, "<td>$(html_escape(cell))</td>")
-                end
-            else
-                write(io, "<td><em>#undef</em></td>")
-            end
+
+    # Check if the user wants to display a summary about the DataFrame that is
+    # being printed. This will be shown using the `title` option of
+    # `pretty_table`.
+    if summary
+        if isempty(title)
+            title = Base.summary(df)
         end
-        write(io, "</tr>")
+    else
+        title = ""
     end
-    if size(df, 1) > mxrow
-        write(io, "<tr>")
-        write(io, "<th>&vellip;</th>")
-        for column_name in cnames
-            write(io, "<td>&vellip;</td>")
+
+    # If `rowid` is not `nothing`, then we are printing a data row. In this
+    # case, we will add this information using the row name column of
+    # PrettyTables.jl. Otherwise, we can just use the row number column.
+    if (rowid === nothing) || (ncol(df) == 0)
+        show_row_number::Bool = get(kwargs, :show_row_number, true)
+        row_labels = nothing
+
+        # If the columns with row numbers is not shown, then we should not
+        # display a vertical line after the first column.
+        vlines = fill(1, show_row_number)
+    else
+        nrow(df) != 1 &&
+            throw(ArgumentError("rowid may be passed only with a single row data frame"))
+
+        # In this case, if the user does not want to show the row number, then
+        # we must hide the row name column, which is used to display the
+        # `rowid`.
+        if !get(kwargs, :show_row_number, true)
```

Comment:
can we add test when row numbers are turned off?

Reply:
Done!